website/docs/guides/getting-started.html.markdown

```diff
@@ -0,0 +1,592 @@
+---
+
+# Getting Started with Kubernetes provider
+
+## Kubernetes
+
+[Kubernetes](https://kubernetes.io/) (K8S) is an open-source workload scheduler
+with focus on containerized applications.
+
+There are at least 2 steps involved in scheduling your first container
+on a K8s cluster. You need the K8S cluster with all its components
+running _somewhere_ and then schedule the K8S resources, like Pods,
+Replication Controllers, Services etc.
+
+This guide focuses mainly on the latter part and expects you to have
+a properly configured & running Kubernetes cluster.
+
+The guide also expects you to run the cluster on a cloud provider
+where K8S can automatically provision a load balancer.
+
+## Why Terraform?
+
+While you could use `kubectl` or similar CLI-based tools mapped to API calls
+to manage all K8S resources described in YAML files,
+orchestration with Terraform presents a few benefits.
+
+ - [HCL](https://www.terraform.io/docs/configuration/syntax.html) - same
+    language for both lower layer with underlying infrastructure (compute)
+    and scheduling layer
+ - drift detection - `terraform plan` will always present you the difference
+    between reality at a given time and config you intend to apply.
+ - full lifecycle management - Terraform doesn't just initially create resources,
+    but offers a single command for creation, update, and deletion of tracked
+    resources without needing to inspect the API to identify those resources.
+ - synchronous feedback - While asynchronous behaviour is often useful,
+    sometimes it's counter-productive as the job of identifying operation result
+    (failures or details of created resource) is left to the user. e.g. you don't
+    have IP/hostname of load balancer until it has finished provisioning,
+    hence you can't create any DNS record pointing to it.
+ - [graph of relationships](https://www.terraform.io/docs/internals/graph.html) -
+    Terraform understands relationships between resources which may help
+    in scheduling - e.g. if a Persistent Volume Claim claims space from
+    a particular Persistent Volume Terraform won't even attempt to create
+    the PVC if creation of the PV has failed.
+
+## Provider Setup
+
+The easiest way to configure the provider is by creating/generating a config
+in a default location (`~/.kube/config`). That allows you to leave the
+provider block completely empty.
+
+```hcl
+provider "kubernetes" {}
+```
+
+If you wish to configure the provider statically you can do so
+
+```hcl
+provider "kubernetes" {
+  host     = "https://104.196.242.174"
+  username = "ClusterMaster"
+  password = "MindTheGap"
+
+  client_certificate     = "${file("~/.kube/client-cert.pem")}"
+  client_key             = "${file("~/.kube/client-key.pem")}"
+  cluster_ca_certificate = "${file("~/.kube/cluster-ca-cert.pem")}"
+}
+```
+
+After specifying the provider we may now run the following command
+to pull down the latest version of the Kubernetes provider.
```

Comment:
I'd suggest "download" rather than "pull down" here, just because it feels less like an idiom.